### ISSUE TITLE
Rake dependency missing for older rubies

### ISSUE THREAD

**Issue by joerg** [test-kitchen/busser-serverspec#28]
The serverspec runner requires rake [1] which is not part of ruby on systems like Centos 7. I just ran some tests, not using chef, and stumbled over that problem:

```
       Transferring files to <default-centos-7>
-----> Running serverspec test suite
-----> Installing Serverspec..
Fetching: rspec-support-3.2.2.gem (100%)
Fetching: rspec-core-3.2.3.gem (100%)
Fetching: diff-lcs-1.2.5.gem (100%)
Fetching: rspec-expectations-3.2.1.gem (100%)
Fetching: rspec-mocks-3.2.1.gem (100%)
Fetching: rspec-3.2.0.gem (100%)
Fetching: rspec-its-1.2.0.gem (100%)
Fetching: multi_json-1.11.0.gem (100%)
Fetching: net-ssh-2.9.2.gem (100%)
Fetching: net-scp-1.2.1.gem (100%)
Fetching: specinfra-2.34.4.gem (100%)
Fetching: serverspec-2.17.0.gem (100%)
-----> serverspec installed (version 2.17.0)
       /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:55:in `require': cannot load such file -- rake (LoadError)
       	from /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:55:in `require'
       
       !!!!!! Ruby Script [/tmp/verifier/gems/gems/busser-serverspec-0.5.6/lib/busser/runner_plugin/../serverspec/runner.rb /tmp/verifier/suites/serverspec] exit code was 1
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: Failed to complete #verify action: [Expected process to exit with [0], but received '1'
---- Begin output of docker exec -t f3ced5c6ed9537bf5279a07596d9c7e6292c7fc551797c563f902cbb441f8b34 sh -c '

BUSSER_ROOT="/tmp/verifier"; export BUSSER_ROOT
GEM_HOME="/tmp/verifier/gems"; export GEM_HOME
GEM_PATH="/tmp/verifier/gems"; export GEM_PATH
GEM_CACHE="/tmp/verifier/gems/cache"; export GEM_CACHE

sudo -E /tmp/verifier/bin/busser test
' ----
STDOUT: -----> Running serverspec test suite
-----> Installing Serverspec..
Fetching: rspec-support-3.2.2.gem (100%)
Fetching: rspec-core-3.2.3.gem (100%)
Fetching: diff-lcs-1.2.5.gem (100%)
Fetching: rspec-expectations-3.2.1.gem (100%)
Fetching: rspec-mocks-3.2.1.gem (100%)
Fetching: rspec-3.2.0.gem (100%)
Fetching: rspec-its-1.2.0.gem (100%)
Fetching: multi_json-1.11.0.gem (100%)
Fetching: net-ssh-2.9.2.gem (100%)
Fetching: net-scp-1.2.1.gem (100%)
Fetching: specinfra-2.34.4.gem (100%)
Fetching: serverspec-2.17.0.gem (100%)
-----> serverspec installed (version 2.17.0)
/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:55:in `require': cannot load such file -- rake (LoadError)
	from /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:55:in `require'
	from /tmp/verifier/gems/gems/busser-serverspec-0.5.6/lib/busser/runner_plugin/../serverspec/runner.rb:20:in `<main>'
!!!!!! Ruby Script [/tmp/verifier/gems/gems/busser-serverspec-0.5.6/lib/busser/runner_plugin/../serverspec/runner.rb /tmp/verifier/suites/serverspec] exit code was 1
STDERR: 
---- End output of docker exec -t f3ced5c6ed9537bf5279a07596d9c7e6292c7fc551797c563f902cbb441f8b34 sh -c '

BUSSER_ROOT="/tmp/verifier"; export BUSSER_ROOT
GEM_HOME="/tmp/verifier/gems"; export GEM_HOME
GEM_PATH="/tmp/verifier/gems"; export GEM_PATH
GEM_CACHE="/tmp/verifier/gems/cache"; export GEM_CACHE

sudo -E /tmp/verifier/bin/busser test
' ----
Ran docker exec -t f3ced5c6ed9537bf5279a07596d9c7e6292c7fc551797c563f902cbb441f8b34 sh -c '

BUSSER_ROOT="/tmp/verifier"; export BUSSER_ROOT
GEM_HOME="/tmp/verifier/gems"; export GEM_HOME
GEM_PATH="/tmp/verifier/gems"; export GEM_PATH
GEM_CACHE="/tmp/verifier/gems/cache"; export GEM_CACHE

sudo -E /tmp/verifier/bin/busser test
' returned 1]
>>>>>> ----------------------
>>>>>> Please see .kitchen/logs/kitchen.log for more details
>>>>>> Also try running `kitchen diagnose --all` for configuration
```

My kitchen.yml has "ruby_bindir: /usr/bin" and in my docker container ruby is:
```
[root@f3ced5c6ed95 /]# ruby --version
ruby 2.0.0p598 (2014-11-13) [x86_64-linux]
```

Installing rake on the system "yum install rubygems-rake", unsetting GEM_PATH und GEM_CACHE and running "/tmp/verifier/gems/bin/busser test" works perfectly fine.

EDIT: Also happens with Fedora 22, so it seems to be a general problem with Rhel derivates. It works perfectly fine on Debian 8 and Ubuntu 15.04.

EDIT2: I just tried minitest which gives the same results. Working on Debian/Ubuntu. Fails on Fedora/CentOS

```
STDOUT: -----> Running minitest test suite
/usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54:in `require': cannot load such file -- rake/testtask (LoadError)
	from /usr/share/rubygems/rubygems/core_ext/kernel_require.rb:54:in `require'
	from /tmp/verifier/gems/gems/busser-minitest-0.3.0/lib/busser/runner_plugin/../minitest/runner.rb:20:in `<main>'
!!!!!! Ruby Script [/tmp/verifier/gems/gems/busser-minitest-0.3.0/lib/busser/runner_plugin/../minitest/runner.rb /tmp/verifier/suites/minitest] exit code was 1
```

[1] https://github.com/test-kitchen/busser-serverspec/blob/master/lib/busser/serverspec/runner.rb#L20

**Comment by cl-lab-k** [comment]
you can put `gem 'rake'` into optional Gemfile:

https://github.com/test-kitchen/busser-serverspec#-usage
https://github.com/test-kitchen/busser-serverspec#-specify-serverspec-version

please file busser-minitest's issue to their tracker.

**Comment by cl-lab-k** [comment]
you can use `shell verifer` instead of `busser`, it lets `serverspec` run on local system.

* https://github.com/test-kitchen/test-kitchen/pull/741 (it has been merged TK 1.5)
* https://rubygems.org/gems/kitchen-verifier-shell (if you use TK 1.4, install gem)

here is my `shell verifer` and `serverspec` example:
https://github.com/cl-lab-k/apache2-take/tree/shell_verifier

I have no plan to make any drastic changes to `busser-serverspec`.
And I think `shell-verifier` + `serverspec` is better choice for testing cookbooks than `busser` verifier + `busser-serverspec`.
So I am preparing the document for translation from`busser` verifier + `busser-serverspec` to `shell-verifier` + `serverspec`.
If the document is prepared, I will put link to it in README.

**Comment by cl-lab-k** [comment]
https://github.com/test-kitchen/busser-serverspec#-notice
